Library/Homebrew/dev-cmd/pr-pull.rb

```diff
@@ -62,6 +63,11 @@ def pr_pull_args
       flag   "--bintray-mirror=",
              description: "Use the specified Bintray repository to automatically mirror stable URLs "\
                           "defined in the formulae (default: `mirror`)."
+      comma_array "--workflows=",
+                  description: "Retrieve artifacts from the specified workflow (default: `tests.yml`) "\
+                               "Comma-separated list to include multiple workflows."
+      comma_array "--ignore-missing-artifacts=",
```

Comment:
Can you explain what this is used for in this case? I don't really fully understand.

Reply:
Yes, sorry it's a little bit hard to follow and we did not come up with a better flag name.

The wheezy_tests.yml workflow is only run for 2 files, patchelf and binutils. Else it is not run at all.

This means that we want to write the following in homebrew-core:
`brew pr-pull --workflows=tests.yml,wheezy_tests.yml --ignore-missing-artifacts=wheezy_tests.yml`

This means: fetch the bottles for both workflows, but if the second one is not run no need to fetch the bottles.

There is also a final check in `get_artifact_url` if the workflow has run (in case you pull a running or failed workflow).

The whole setup is a little bit weird but it is because we can not have conditionals in the GitHub workflow .yml files.
I tried to keep the code as generic as possible.

There will soon be a third workflow for linux bottles, and maybe a fourth for self-hosted runners.

Reply:
> This means that we want to write the following in homebrew-core:
> `brew pr-pull --workflows=tests.yml,wheezy_tests.yml --ignore-missing-artifacts=wheezy_tests.yml`

Is this run by a human or @BrewTestBot? If a human: it feels like having a dedicated flag for all this is going to be less code complexity and less error-prone than remembering the exact invocation.

> There will soon be a third workflow for linux bottles, and maybe a fourth for self-hosted runners.

We'll need to figure out a way of merging this workflows, I think.

Reply:
By the bot, here: https://github.com/Homebrew/homebrew-core/blob/master/.github/workflows/publish-commit-bottles.yml#L41

> We'll need to figure out a way of merging this workflows, I think.

I already discussed this with @dawidd6 and there seem to be no easy solution yet. We might have some code duplication in workflows for some time.

Reply:
I am seeing all this as a sort of V1, to get things going. There are still things to iron out, but if I remember well it took a few months to get the current workflows setup as we have them now. Things will evolve (I hope).